crates/ruff_linter/src/rules/flake8_quotes/rules/avoidable_escaped_quote.rs

```diff
@@ -0,0 +1,258 @@
+                if let Some(fstring_context) = fstrings.last_mut() {
+                    fstring_context.do_not_check_for_escaped_quote();
+                    continue;
+                }
+            }
+        }
+
+        match tok {
+            Tok::String {
+                value: string_contents,
+                kind,
+                triple_quoted,
+            } => {
+                // Check if we're using the preferred quotation style.
+                let uses_preferred_quote = leading_quote(locator.slice(tok_range))
+                    .is_some_and(|text| text.contains(quotes_settings.inline_quotes.as_char()));
+                if kind.is_raw() || *triple_quoted || !uses_preferred_quote {
+                    continue;
+                }
+
+                if string_contents.contains(quotes_settings.inline_quotes.as_char())
+                    && !string_contents.contains(quotes_settings.inline_quotes.opposite().as_char())
+                {
+                    let mut diagnostic = Diagnostic::new(AvoidableEscapedQuote, tok_range);
+                    if settings.rules.should_fix(diagnostic.kind.rule()) {
+                        let fixed_contents = format!(
+                            "{prefix}{quote}{value}{quote}",
+                            prefix = kind.as_str(),
+                            quote = quotes_settings.inline_quotes.opposite().as_char(),
+                            value = unescaped_string(string_contents)
+                        );
+                        diagnostic.set_fix(Fix::automatic(Edit::range_replacement(
+                            fixed_contents,
+                            tok_range,
+                        )));
+                    }
+                    diagnostics.push(diagnostic);
+                }
+            }
+            Tok::FStringStart => {
+                let text = locator.slice(tok_range);
+                // Check for escaped quote only if we're using the preferred quotation
+                // style and it isn't a triple-quoted f-string.
+                let check_for_escaped_quote = text
+                    .contains(quotes_settings.inline_quotes.as_char())
+                    && !is_triple_quote(text);
+                fstrings.push(FStringContext::new(check_for_escaped_quote, tok_range));
+            }
+            Tok::FStringMiddle {
+                value: string_contents,
+                is_raw,
+            } if !is_raw => {
+                let Some(context) = fstrings.last_mut() else {
+                    continue;
+                };
+                if !context.check_for_escaped_quote {
+                    continue;
+                }
+                if string_contents.contains(quotes_settings.inline_quotes.as_char())
+                    && !string_contents.contains(quotes_settings.inline_quotes.opposite().as_char())
+                {
+                    context.push_fstring_middle_range(tok_range);
```

Comment:
Nit: The name of the method is misleading (or the field name is). It doesn't track **all** fstring middle parts. It tracks the fstring middle parts that contain the preferred quotes only.